logr.go

```diff
@@ -269,6 +289,32 @@ func (l Logger) WithCallDepth(depth int) Logger {
 	return l
 }
 
+// Helper returns a Logger instance that skips the direct caller when
+// logging call site information, if possible.  This is useful for
+// users who have helper functions between the "real" call site and
+// the actual calls to Logger methods.
+//
```

Comment:
I think we should spell out that, in particular, implementations that build on `testing.T` need this.